### PR TITLE
Issue #1319: Keep comments when loading and saving plys 

### PR DESCRIPTION
**Pull request by eduardoiscar**
Hi Daniel, 

I have implemented the ability to save the ply comments as metadata attached to the pointcloud during loading that gets written back to the ply file when saving it following your recommendations.

Thanks a lot for the support

### CODE REVIEW

File: libs/qCC_io/src/PlyFilter.cpp [hunk 1]
```diff
@@ -379,8 +379,21 @@ CC_FILE_ERROR PlyFilter::saveToFile(ccHObject* entity, QString filename, e_ply_s
...
 	}
 	
-	ply_add_comment(ply, qPrintable(FileIO::createdBy()));
-	ply_add_comment(ply, qPrintable(FileIO::createdDateTime()));
+	if (vertices->hasMetaData("ply.comments")) 
+	{
+		QStringList comments = vertices->getMetaData("ply.comments").toStringList();
+		for(int i = 0; i<comments.size(); i++)
+		{
+			ply_add_comment(ply, comments[i].toStdString().c_str());
```
Comment (dgirardeau) [inline libs/qCC_io/src/PlyFilter.cpp:387]: I would use the qPrintable() shortcut instead

File: libs/qCC_io/src/PlyFilter.cpp [hunk 2]
```diff
@@ -379,8 +379,21 @@ CC_FILE_ERROR PlyFilter::saveToFile(ccHObject* entity, QString filename, e_ply_s
 		else result = 0;
 	}
 	
-	ply_add_comment(ply, qPrintable(FileIO::createdBy()));
-	ply_add_comment(ply, qPrintable(FileIO::createdDateTime()));
+	if (vertices->hasMetaData("ply.comments")) 
+	{
+		QStringList comments = vertices->getMetaData("ply.comments").toStringList();
+		for(int i = 0; i<comments.size(); i++)
```
Comment (dgirardeau) [inline libs/qCC_io/src/PlyFilter.cpp:385]: Maybe use the modern 'for':
```
for (QString comment : comments)
{
    ply_add_comment(ply, qPrintable(comment));
}
```

File: libs/qCC_io/src/PlyFilter.cpp [hunk 3]
```diff
@@ -917,13 +930,15 @@ CC_FILE_ERROR PlyFilter::loadFile(const QString& filename, const QString& inputT
 	/******************/
 	/***  Comments  ***/
 	/******************/
+	QStringList comments = QStringList();
```
Comment (dgirardeau) [inline libs/qCC_io/src/PlyFilter.cpp:933]: No need to initialize it:
```
QStringList comments;
```

**Review by dgirardeau — CHANGES_REQUESTED** [review]
Thanks for the fast implementation. See my comments.

**Comment by eduardoiscar** [comment]
Ok, made the suggested changes.

**Review by dgirardeau — APPROVED** [review]
Thanks!

**Pull request merged**